python_modules/libraries/dagster-slack/dagster_slack/sensors.py

```diff
@@ -183,3 +201,108 @@ def slack_on_run_failure(context: RunFailureSensorContext):
         slack_client.chat_postMessage(channel=channel, blocks=blocks, text=main_body_text)
 
     return slack_on_run_failure
+
+
+def _default_freshness_message_text_fn(context: FreshnessPolicySensorContext) -> str:
+    return f"Asset `{context.asset_key.to_user_string()}` is now {context.minutes_late:.2f} minutes late."
+
+
+@experimental
+def make_slack_on_freshness_policy_status_change_sensor(
+    channel: str,
+    slack_token: str,
+    asset_selection: AssetSelection,
+    warn_after_minutes_late: float = 0,
+    notify_when_on_time: bool = False,
+    text_fn: Callable[[FreshnessPolicySensorContext], str] = _default_freshness_message_text_fn,
+    blocks_fn: Optional[Callable[[FreshnessPolicySensorContext], List[Dict[Any, Any]]]] = None,
+    name: Optional[str] = None,
+    dagit_base_url: Optional[str] = None,
+    default_status: DefaultSensorStatus = DefaultSensorStatus.STOPPED,
+):
+    """Create a that will message the given Slack channel whenever an asset in the provided
+    AssetSelection becomes out of date. Messages are only fired when the state changes, meaning
+    only a single slack message will be sent (when the asset begins to be out of date). If
+    `notify_when_on_time` is set to `True`, a second slack message will be sent once the asset
+    is on time again.
+
+    Args:
+        channel (str): The channel to send the message to (e.g. "#my_channel")
+        slack_token (str): The slack token.
+            Tokens are typically either user tokens or bot tokens. More in the Slack API
+            documentation here: https://api.slack.com/docs/token-types
+        asset_selection (AssetSelection): The selection of assets which this sensor will monitor.
+            Alerts will only be fired for assets that have a FreshnessPolicy defined.
+        warn_after_minutes_late (float): How many minutes past the specified FreshnessPolicy this
+            sensor will wait before firing an alert (by default, an alert will be fired as soon as
+            the policy is violated).
+        notify_when_on_time (bool): If a success message should be sent when the asset becomes on
```

Comment:
Maybe change this to `notify_when_back_on_time` or `notify_when_no_longer_late`?